server/usecases/board/list.go

```diff
@@ -0,0 +1,168 @@
+package board
+
+import (
+	"database/sql"
+
+	"github.com/h3poteto/fascia/lib/modules/database"
+	"github.com/h3poteto/fascia/lib/modules/logging"
+	"github.com/h3poteto/fascia/server/domains/list"
+	"github.com/h3poteto/fascia/server/domains/repo"
+	"github.com/h3poteto/fascia/server/domains/task"
+	repository "github.com/h3poteto/fascia/server/infrastructures/list"
+)
+
+// InjectDB set DB connection from connection pool.
+func InjectDB() *sql.DB {
+	return database.SharedInstance().Connection
+}
+
+func InjectListRepository() list.Repository {
+	return repository.New(InjectDB())
+}
+
+func ListHasCloseAction(list *list.List) (bool, error) {
+	return list.HasCloseAction()
+}
+
+func FindList(projectID, listID int64) (*list.List, error) {
+	return list.Find(projectID, listID, InjectListRepository())
+}
+
+func CreateList(projectID, userID int64, title, color string, optionID sql.NullInt64, isHidden bool) (*list.List, error) {
+	nullableTitle := sql.NullString{String: title, Valid: true}
+	nullableColor := sql.NullString{String: color, Valid: true}
+	l := list.New(0, projectID, userID, nullableTitle, nullableColor, optionID, isHidden, InjectListRepository())
+	err := l.Create(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	go func(l *list.List) {
+		projectID := l.ProjectID
+		p, err := FindProject(projectID)
+		// TODO: log
+		if err != nil {
+			return
+		}
+		token, err := p.OauthToken()
+		if err != nil {
+			return
+		}
+		repo, err := p.Repository(InjectRepoRepository())
+		if err != nil {
+			return
+		}
+		err = fetchCreatedList(l, token, repo)
+		if err != nil {
+			return
+		}
+	}(l)
+	return l, nil
+}
+
+func fetchCreatedList(l *list.List, oauthToken string, repo *repo.Repo) error {
+	if repo != nil {
+		label, err := repo.CheckLabelPresent(oauthToken, l.Title.String)
+		if err != nil {
+			return err
+		} else if label == nil {
+			label, err = repo.CreateGithubLabel(oauthToken, l.Title.String, l.Color.String)
+			if err != nil {
+				return err
+			}
+		} else {
+			// 色だけはこちら指定のものに変更したい
+			_, err := repo.UpdateGithubLabel(oauthToken, l.Title.String, l.Title.String, l.Color.String)
+			if err != nil {
+				return err
+			}
+			logging.SharedInstance().MethodInfo("list", "Save").Info("github label already exist")
+		}
+	}
+	return nil
+}
+
+func UpdateList(l *list.List, title, color string, optionID int64) (*list.List, error) {
```

Comment:
__[Golint]__ exported function UpdateList should have comment or be unexported ([link](https://sider.review/gh/repos/50192141/pulls/454#issue-3810552))

<sub>posted by Sider</sub>